prov/gni/src/gnix_auth_key.c

```diff
@@ -36,13 +36,87 @@
 
 #include "gnix_auth_key.h"
 #include "gnix_hashtable.h"
+#include "gnix_shmem.h"
+#include "gnix_bitmap.h"
 #include "gnix.h"
 
 #define GNIX_AUTH_KEY_HASHSEED 0xdeadbeef
 
+#define GAI_VERSION(major, minor) (((major) << 16) + (minor))
+typedef enum gnix_global_auth_info_version {
+	GNIX_GAI_VERSION_1 = GAI_VERSION(1, 0), /* initial version */
+} gnix_global_auth_info_version_t;
+
+#define MAX_VMDH_TAGS 4
+#define MAX_VMDH_REGS 4096
+#define BITMAP_ARR_SIZE(nbits) \
+	(GNIX_BITMAP_BLOCKS(nbits) * sizeof(gnix_bitmap_block_t))
+#define GNIX_VMDH_PATH "/tmp/gnix_vmdh_info"
```

Comment:
> How is this suppose to work if there are multiple jobs on a node, i.e. slurm with non exclusive mode?

If you don't have multiple jobs sharing the same file, there will be issues with reporting useful feedback to the application. I need to modify the permissions to allow all users to edit the file, because otherwise it won't work with the MAMU case

> I also don't think its a good idea to hardware a filename that is opened in /tmp. At least something that honors $TMPDIR environment variable.

I can add an environment variable, but I'll add the cautionary warning here: If the environment is propagated uniformly, there could be issues -- However, that isn't our issue to resolve, so it's a minor thing.